src/rpc_server.rs

```diff
@@ -3129,6 +3234,43 @@ impl RPC for NeptuneRPCServer {
         Ok(Self::cpu_temp_inner())
     }
 
+    // documented in trait. do not add doc-comment.
+    async fn pow_challenge(
+        self,
+        _context: tarpc::context::Context,
+        token: rpc_auth::Token,
+    ) -> RpcResult<Option<GuessChallengeInfo>> {
+        log_slow_scope!(fn_name!());
+        token.auth(&self.valid_tokens)?;
+
+        let Some(proposal) = self
+            .state
+            .lock_guard()
+            .await
+            .block_proposal
+            .map(|x| x.to_owned())
+        else {
+            return Ok(None);
+        };
```

Comment:
```suggestion
        let state = self.state.lock_guard().await;
        let Some(proposal) = state.block_proposal.map(|x| x.to_owned()) else {
            return Ok(None);
        };
```

Reply:
again, be mindful that the suggested change is not dropping the acquired lock.  I haven't checked subsequent code, so maybe it's ok, but it is a logic change.